radix-engine/src/kernel/actor.rs

```diff
@@ -9,23 +10,26 @@ pub struct InstanceContext {
     pub outer_blueprint: String,
 }
 
-/// No method acting here!
 #[derive(Debug, Clone, PartialEq, Eq, ScryptoSbor)]
-pub struct MethodActor {
-    pub global_address: Option<GlobalAddress>,
+pub struct RuntimeReceiverInfo {
     pub node_id: NodeId,
     pub module_id: ObjectModuleId,
-    pub module_object_info: ObjectInfo,
+    pub is_direct_access: bool,
+    pub object_info: ObjectInfo,
+}
 
+#[derive(Debug, Clone, PartialEq, Eq, ScryptoSbor)]
+pub struct MethodActor {
     pub ident: String,
+    pub receiver_info: RuntimeReceiverInfo,
+    pub global_address: Option<GlobalAddress>,
```

Comment:
I've also refactored `MethodActor` a bit in https://github.com/radixdlt/radixdlt-scrypto/pull/1246/files#diff-a1773c84b7bd8f3422fd46bd349a608119e9beb86872c8e1f70c1a04ef4d3ecdL14

Could we revert these cleanups?

Reply:
Briefly looked at the other PR. I'm not sure I agree with that refactor.

I'm happy to revert any specific change here, if it doesn't look right to you.

Reply:
Yup, I think just removing `RuntimeReceiverInfo` and having all info flat in `MethodActor` for now would make it easiest. For example, I consider `object_info` as not required but is essentially just a cache of this info for `MethodActor`.

Reply:
Yes, removed it.